go/pserver/service.go

```diff
@@ -52,14 +67,34 @@ type Service struct {
 	optMap map[string]*optimizer
 }
 
+type checkpoint struct {
+	Uuid      string
+	Md5sum    string
+	Timestamp string
+}
+
+//serialize ParameterWithConfig to byte stream
+func GetBytes(content ...interface{}) ([]byte, error) {
```

Comment:
Making `content ...interface{}` adds more complexity to the code: since it's interface type that we need to encode, we have to call `gob.Register`. It's harder to understand the code (people need to search for what does `gob.Register` do. And it's harder to maintain the code (whenever adds a new type for `GetBytes` to use, maintainer need to add `gob.Register` as well, it's hard to track.

Since here we only need to call `GetBytes` twice, and this function does not have much code. Maybe just put it inline? (and remove `gob.Register`)